share/trace.go

```diff
@@ -36,7 +36,8 @@ func (s *metadataSupplier) Keys() []string {
 }
```

Comment:
The code is overall well-written, but there are a few changes that I would recommend:

1. Error Handling: In the `Extract` and `Inject` functions, you're asserting that the context value is a map of strings. If it's not, this will cause a panic. You should add error handling to ensure that the value is of the expected type.

```go
meta, ok := value.(map[string]string)
if !ok {
    // handle error or return
}
```

2. Unnecessary Code: In the `Extract` and `Inject` functions, you're creating a new map if `meta` is `nil` and then setting it in the context. However, you're not using this new map anywhere else in these functions. Consider removing these lines unless you have a specific reason for them.

3. Code Duplication: The code to create a new map and set it in the context if `meta` is `nil` is duplicated in `Extract` and `Inject`. Consider moving this to a separate function to reduce code duplication.

4. Commenting: Add comments to your code, especially to exported types and functions. This will make it easier for others to understand what each part of your code is doing.

5. Naming: The `metadataSupplier` struct could have a more descriptive name. Consider renaming it to something that better describes its purpose, like `MetadataCarrier`.

6. Type Assertion: In the `Extract` function, you're asserting that `meta` is a map of strings without checking if the assertion is true. This could cause a panic if `meta` is not a map of strings. Add a check to ensure that the type assertion is successful.

```go
meta, ok := meta.(map[string]string)
if !ok {
    // handle error or return
}
``